### PR TITLE
Have App Progress Through Chapters

### PR DESCRIPTION
**Pull request by michaelwehrley**
Have Chapters Progress

* Add timeout to progress through chapters.
* Reset timeout after each new chapter is viewed.
* Clear timeouts if a user is viewing the second page of a chapter.
* Add specs.
* Change `BASE_URL` to staging url.
* Rename `Things` to more meaningful `Resource`.

[Finishes: #93560130]

### CODE REVIEW

File: src/js/controllers/PagesController.js [hunk 1]
```diff
@@ -53,9 +80,27 @@
...
   }
 
   angular
+  .module('ednavigator.constant')
+  .constant('INTERVAL', {
+    HALF_SECOND: 500,
+    TEN_SECONDS: 5000
+  })
+  .constant('PASSWORD', {
+    CODE: '1111'
```
Comment (ericcf) [inline src/js/controllers/PagesController.js]: I suggest making this value configurable.

File: src/js/services/TimeoutPromise.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* Builds a Page */
+(function() {
+  'use strict';
+
+  function TimeoutPromise() {
+    return {};
```
Comment (ericcf) [inline src/js/services/TimeoutPromise.js]: ?

Comment (michaelwehrley) [inline src/js/services/TimeoutPromise.js]: I need this hash to persist as a service would allow for.

**Comment by michaelwehrley** [comment]
@ericcf I was able to rebase this and it is all set.  Thanks!

**Pull request merged**